### DISCUSSION TITLE
Questions about buttons.

### DISCUSSION THREAD

**Discussion by jtreagan** [fltk-rs/fltk-rs#1593] [Q&A]
I've been experimenting with Buttons, trying to get a handle on how they work and how to use them.  So, this discussion thread is intended to serve as a place to ask questions that come up while I'm experimenting.  It might also serve to help some other FLTK newbie work through how to use buttons.

I've focused some on Radio buttons because I need to use them in the application I'm working on.  I've found that getting them to use an assigned size just doesn't work.  It finally dawned on me that the size of Radio buttons seems to be tied to the size of the Flex box they live in.  It certainly seems that way as I've been experimenting with them.

**Question 1:**  If you place a Radio button inside a Flex box, is the size of the button always tied to the size of the Flex box?   As a followup to that, is this "match the size of the flex box"  rule true of most of the alternate button structs -- like `ReturnButton` or `ShortcutButton`?

**Comment by MoAlyousef** [comment]
This is true for all Flex child widgets. If a flex has a single child, it occupies the whole flex, if 2 children, each occupies half etc. Flex offers a `fixed` method which specifies the size of the child widget.
```rust
use fltk::{prelude::*, *};

fn main() {
    let a = app::App::default();
    let mut win = window::Window::default().with_size(100, 200);
    let mut col = group::Flex::default_fill().column();
    let b1 = button::Button::default().with_label("Inc");
    col.fixed(&b1, 30);
    frame::Frame::default();
    let b2 = button::Button::default().with_label("Dec");
    col.fixed(&b2, 30);
    col.end();
    win.end();
    win.show();
    a.run().unwrap();
}
```
Without the call to fixed, each of the column's children will occupy a third of the column. The calls to fixed for the first and last widgets means the middle widget gets to expand to fill the remaining column space.

**Reply by jtreagan** [reply]
Thanks, @MoAlyousef.  I played around with your code a bit.  Added a couple of buttons without calling `fixed()` on them.  They showed up with the same width as the other buttons with a deeper height, but did **not** fill all the remaining space as you suggested above.  I'm pretty sure I'll be making use of `fixed()` in the future.

**Reply by MoAlyousef** [reply]
Most welcome. A Flex has a default spacing of 5. This can be changed using Flex::set_spacing.
